teabot/worker.py

```diff
@@ -25,8 +33,8 @@ def do_work():
     server_link.send_queued_update_if_time()
     global last_status, last_number_of_cups
 
-    current_weight = weight_sensor.get_reading()
-    temperature = temperature_sensor.get_reading()
+    current_weight = weight_sensor.get_readings(wait=True)[-1]['reading']
```

Comment:
You're right: get_readings() was doing far too much.

Since we now have some methods that read the underlying sensor and some that just get saved values, I have done a bit of refactoring + renaming.